lang/wasm.cc

```diff
@@ -218,7 +259,21 @@ struct from_val_visitor {
             return bytes_opt{};
         }
         data += sizeof(int32_t); // size of the return type was consumed
-        return t.decompose(t.deserialize(bytes_view(reinterpret_cast<int8_t*>(data), ret_size)));
+        bytes_opt ret = t.decompose(t.deserialize(bytes_view(reinterpret_cast<int8_t*>(data), ret_size)));
+        auto free_export = instance.get(store, "_scylla_free");
+        if (free_export) {
+            std::vector<wasmtime::Val> free_argv;
+            free_argv.push_back(val.i32());
+            wasmtime::Func* free_func = std::get_if<wasmtime::Func>(&*free_export);
+            if (!free_func) {
+                throw wasm::exception(format("Exported object {} is not a function", "_scylla_free"));
+            }
+            auto result = free_func->call(store, free_argv);
+            if (!result) {
+                throw wasm::exception("Calling wasm function failed: " + result.err().message());
+            }
+        }
+        return ret;
```

Comment:
You should also free the arguments.

Reply:
As we've decided, we're passing the burden of freeing the arguments to the wasm program, but we will handle that by ourself using a macro

Reply:
The documentation should note that. Also who is responsible for freeing the return value.